### PR TITLE
Feature/cu kk7a7m player controller navigation

### PR DESCRIPTION
**Pull request by Yonimevicio**
- Animation bugs fixes 
- Code improvements
- NavMesh implemented
- New dash implemented
- Merged with trail -> Onimaru won't shoot trail but I implemented it while we are waiting for its shoot implementation.
- Test scene -> PlayerNavigationTest

### CODE REVIEW

**Comment by jcarlos0305** [comment]
Task linked: [CU-kk7a7m Player controller implement navigation movement](https://app.clickup.com/t/kk7a7m)

File: Penteract/Assets/Scripts/PlayerController.cpp [hunk 1]
```diff
@@ -97,15 +97,21 @@ void PlayerController::Start() {
...
+	aux = GameplaySystems::GetGameObject(dashAudioSourceUID);
+	if (aux) {
 		dashAudioSource = aux->GetComponent<ComponentAudioSource>();
 	}
 	firstTime = true;
+
+	agent = GetOwner().GetComponent<ComponentAgent>();
+	if (agent) {
+		agent->SetMaxSpeed(fangMovementSpeed);
+		agent->SetMaxAcceleration(9999);
```
Comment (magalenyo) [inline Penteract/Assets/Scripts/PlayerController.cpp:113]: Could be good to have it as a constant or macro

File: Penteract/Assets/Scripts/PlayerController.cpp [hunk 2]
```diff
@@ -164,7 +178,9 @@ void PlayerController::Dash() {
 }
 
 bool PlayerController::CanDash() {
-	return !dashing && !dashInCooldown && fang->IsActive();
+	//TODO: DELETE BEFORE IMPLEMENT DASH WITH NAVMESH
```
Comment (magalenyo) [inline Penteract/Assets/Scripts/PlayerController.cpp:181]: Still a TODO?

**Review by magalenyo — APPROVED** [review]

**Review by jcarlos0305 — DISMISSED** [review]

**Review by Jordi-Gil — DISMISSED** [review]

**Review by isaacfdez — APPROVED** [review]

**Review by jcarlos0305 — DISMISSED** [review]

**Review by isaacfdez — APPROVED** [review]

File: Penteract/Assets/Scripts/PlayerController.cpp [hunk 3]
```diff
@@ -187,51 +210,58 @@ void PlayerController::SwitchCharacter() {
...
+	return !shooting && ((fang->IsActive() && fangTrail) || (onimaru->IsActive() && onimaruTrail));
 }
 
 void PlayerController::Shoot() {
 	ComponentTransform* transform = GetOwner().GetComponent<ComponentTransform>();
-	FangGuntransform = fangGun->GetComponent<ComponentTransform>();
-
+	
 	if (CanShoot()) {
 		if (shootAudioSource) {
```
Comment (magalenyo) [inline Penteract/Assets/Scripts/PlayerController.cpp:229]: I'd recommend externalizing the Shooting functions in a ProjectileController so you just need to p.eg: ProjectileController:Shoot(direction, speed, etc.) and some logic can be externalized into a specific module

**Review by magalenyo — APPROVED** [review]
Check comments

**Review by isaacfdez — DISMISSED** [review]

**Review by isaacfdez — APPROVED** [review]

**Review by Jordi-Gil — APPROVED** [review]

**Review by jcarlos0305 — APPROVED** [review]

**Pull request merged**